crates/bitwarden/src/vault/cipher/cipher.rs

```diff
@@ -0,0 +1,109 @@
+use chrono::{DateTime, Utc};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::crypto::CipherString;
+
+use super::{attachment, card, field, identity, login, password_history, secure_note};
+
+#[derive(Clone, Copy, Serialize, Deserialize, Debug, JsonSchema)]
+pub enum CipherType {
+    Login = 1,
+    SecureNote = 2,
+    Card = 3,
+    Identity = 4,
+}
+
+#[derive(Clone, Copy, Serialize, Deserialize, Debug, JsonSchema)]
+pub enum CipherRepromptType {
+    None = 0,
+    Password = 1,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct Cipher {
```

Comment:
We're missing,

```
organizationUseTotp: boolean;
edit: boolean;
viewPassword: boolean;
localData: LocalData;
```

I assume these are not really needed?

Reply:
They aren't really needed, but I think we should define the model as completely as possible, so I've added them.